ERCS/erc-7835.md

```diff
@@ -0,0 +1,250 @@
+
+- **1.Wallet-level parser interface:** wallet_metaDataAbstractParser
+  A universal entry point that Wallets can use to request metadata parsing.
+- **2.Smart contract-level parsing logic:** parserHandler and parserData
+  Smart contracts that handle specific parsing logic, receiving parameters from Wallets and returning readable metadata.
+
+The aim of this proposal is to establish a general-purpose, trustless, and decentralized metadata parsing standard that enhances user experience, security, and the readability of Ethereum ecosystem metadata.
+
+## Motivation
+
+Wallets serve as the primary interface between users and the Ethereum ecosystem. To improve user experience and security, Wallets need to parse various types of on-chain metadata, such as:
+
+- ENS mappings to addresses.
+- Standard assert's readable label (eg. [ERC-20](./erc-20.md), [ERC-721](./erc-721.md))
+- On-chain content (e.g., resources like eth.limo).
+- Cross Chain intentions standard like [EIP-7683](./eip-7683.md)
+- Other human-readable on-chain information.
+
+Currently, Wallets lack a standardized way of parsing metadata, leading to several challenges:
+
+- **1.Inconvenience:** Different Wallets implement metadata parsing differently, resulting in inconsistent user experiences.
+- **2.Security Risks:** Scattered parsing logic increases the attack surface for malicious actors.
+- **3.Low Readability:** On-chain data is often difficult for users to understand without additional processing.
+
+This proposal aims to solve these problems by creating a standardized, secure, and user-friendly way for Wallets to parse on-chain metadata.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Work Flow
+
+![workflow](../assets/erc-7835/workflow.png)
+
+### 1. Wallet-Level Entry Point: wallet_metaDataAbstractParser
+
+`wallet_metaDataAbstractParser` is a universal interface that Wallets can use to request metadata parsing. It accepts the following parameters:
+
+- `parserHandler:` The address of the smart contract that implements the parsing logic.
+- `parserData:` The raw metadata parameters to be parsed (of type bytes).
+
+```typescript
+export interface IWalletMetaDataAbstractParser {
+  /**
+   * Universal interface for parsing on-chain metadata
+   * @param parserHandler Address of the smart contract handling metadata parsing
+   * @param parserData Raw metadata parameters to be parsed
+   * @returns A promise that resolves to the parsed metadata of generic type T
+   */
+  wallet_metaDataAbstractParser<T>(
+    parserHandler: string, // Address of the smart contract
+    parserData: Uint8Array // Raw metadata parameters as bytes
+  ): Promise<T>; // The return type is a generic type T
+}
+```
```

Comment:
Personally I prefer to see [JSON Schema](https://json-schema.org/) definitions for RPC endpoints because that's the format https://github.com/ethereum/execution-apis uses.